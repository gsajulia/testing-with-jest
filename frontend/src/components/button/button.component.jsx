/* Styles */
import './button.styles.css';

export const Button = ({ buttonValue, handleChange }) => {
    return (
        <div className="each-button-container">
            <button onClick={() => handleChange(buttonValue)} className={`button ${buttonValue}`}>
                <div className="inside-button">
                    {buttonValue !== "" &&  buttonValue !== undefined? 
                        buttonValue[0].toUpperCase() + buttonValue.substr(1) : null}
                </div>
            </button>
        </div>
    )
}