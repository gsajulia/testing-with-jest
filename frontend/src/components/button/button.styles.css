.slytherin {
    background: repeating-linear-gradient(
      45deg,
      #7a7a7a,
      #7a7a7a 3px,
      #0b3a07 3px,
      #0b3a07 9px,
      #7a7a7a 9px,
      #7a7a7a 10px,
      #0b3a07 10px,
      #0b3a07 30px);
  }
  
  .gryffindor {
    background: repeating-linear-gradient(
    45deg,
    #c6a513,
    #c6a513 3px,
    #6d0000 3px,
    #6d0000 9px,
    #c6a513 9px,
    #c6a513 10px,
    #6d0000 10px,
    #6d0000 30px);
  }
  
  .ravenclaw {
    background: repeating-linear-gradient(
      45deg,
      #7a7a7a,
      #7a7a7a 3px,
      #191c68 3px,
      #191c68 9px,
      #7a7a7a 9px,
      #7a7a7a 10px,
      #191c68 10px,
      #191c68 30px);
  }
  
  .hufflepuff {
    background: repeating-linear-gradient(
      45deg,
      #161616,
      #161616 3px,
      #c6a513 3px,
      #c6a513 9px,
      #161616 9px,
      #161616 10px,
      #c6a513 10px);
  }
  
  .button {
    width: 170px;
    height: 60px;
    color: #000;
    border: none;
    padding: 15px 30px;
    cursor: pointer;
    border-radius: 10px;
    border: 1px solid #757575;
  }
  
  .inside-button {
    background-color: #ffe082;
    border-radius: 5px;
    font-weight: bold;
    font-size: 20px;
    border: 2px solid #ffe082;
  }

  .each-button-container {
    padding: 5px;
  }