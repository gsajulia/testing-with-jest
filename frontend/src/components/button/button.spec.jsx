import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Button } from './button.component';

const buttonValue = "gryffindor";

describe('<Button />', () => {
  it('should render the button with the text of the house', () => {
    render(<Button buttonValue={buttonValue}/>);

    expect.assertions(1);

    const button = screen.getByRole('button', { name: "Gryffindor" }); // or name: /gryffindor/i
    expect(button).toBeInTheDocument(); //assertion
  });

  it('should call function on button click', () => {
    const fn = jest.fn();
    render(<Button buttonValue={buttonValue} handleChange={fn} />);

    const button = screen.getByRole('button', { name: "Gryffindor" });
    
    userEvent.click(button); // or fireEvent.click(button);

    expect(fn).toHaveBeenCalled();
  });
});