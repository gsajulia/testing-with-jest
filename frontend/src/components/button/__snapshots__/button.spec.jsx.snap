// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should match snapshot 1`] = `
<div
  class="each-button-container"
>
  <button
    class="button undefined"
  >
    <div
      class="inside-button"
    />
  </button>
</div>
`;
